frontend/apps/filemanager/filemanagerhistory.lua

```diff
@@ -128,45 +132,40 @@ function FileManagerHistory:onMenuHold(item)
                 callback = function()
                     require("readhistory"):removeItem(item)
                     self._manager:updateItemTable()
-                    UIManager:close(self.histfile_dialog)
+                    UIManager:close(dialog)
                 end,
             },
         },
         {
             {
                 text = _("Delete"),
-                enabled = (item.file ~= currently_opened_file and lfs.attributes(item.file, "mode")) and true or false,
+                enabled = file_actions_enabled,
                 callback = function()
-                    UIManager:show(ConfirmBox:new{
-                        text = T(_("Are you sure that you want to delete this document?\n\n%1\n\nIf you delete a file, it is permanently lost."),
-                            BD.filepath(item.file)),
-                        ok_text = _("Delete"),
-                        ok_callback = function()
-                            local FileManager = require("apps/filemanager/filemanager")
-                            FileManager:deleteFile(item.file)
-                            require("readhistory"):fileDeleted(item.file) -- (will update "lastfile" if needed)
-                            self._manager:updateItemTable()
-                            UIManager:close(self.histfile_dialog)
-                        end,
-                    })
+                    local function post_delete_callback()
+                        self._manager:updateItemTable()
+                        self._manager.files_updated = true
+                        UIManager:close(dialog)
+                    end
+                    local FileManager = require("apps/filemanager/filemanager")
+                    FileManager:deleteFileDialog(item.file, post_delete_callback)
                 end,
             },
             {
                 text = _("Book information"),
                 enabled = FileManagerBookInfo:isSupported(item.file),
                 callback = function()
                     FileManagerBookInfo:show(item.file)
-                    UIManager:close(self.histfile_dialog)
+                    UIManager:close(dialog)
                 end,
              },
         },
     }
-    self.histfile_dialog = ButtonDialogTitle:new{
+    dialog = ButtonDialogTitle:new{
         title = BD.filename(item.text:match("([^/]+)$")),
         title_align = "center",
         buttons = buttons,
     }
-    UIManager:show(self.histfile_dialog)
+    UIManager:show(dialog)
```

Comment:
No issue with this renaming from a object property to a local variable, with the following ? is it the same `self.histfile_dialog` object that is accessed, picked from, and replaced ?
https://github.com/koreader/koreader/blob/15fb73384fcb99753fea839bc48f842540a56290/plugins/coverbrowser.koplugin/covermenu.lua#L413-L428

Reply:
You are so right